src/models/account.cpp

```diff
@@ -507,6 +529,181 @@ int Account::importFromCSV(const std::string& path)
     return imported;
 }
 
+int Account::importFromOFX(const std::string& path)
+{
+    int imported{ 0 };
+    unsigned int nextId{ getNextAvailableTransactionId() };
+    std::regex xmlTagRegex{ "^<([^/>]+|/STMTTRN)>(?:[+-])?([^<]+)" };
+    std::smatch xmlTagMatch;
+    Transaction* transaction{ nullptr };
+    bool skipTransaction{ false };
+    std::ifstream file{ path };
+    if (file.is_open())
+    {
+        std::string line;
+        while (getline(file, line))
+        {
+            // Extract tag and content
+            if (std::regex_match(line, xmlTagMatch, xmlTagRegex))
+            {
+                std::string tag{ xmlTagMatch[1].str() };
+                std::string content{ xmlTagMatch[2].str() };
+                size_t found = content.find('\r');
+                if (found != std::string::npos)
+                {
+                    content.erase(found);
+                }
+                // Add previous transaction
+                if (tag == "/STMTTRN" && transaction != nullptr)
+                {
+                    if (!skipTransaction)
+                    {
+                        addTransaction(*transaction);
+                        imported++;
+                    }
+                    skipTransaction = false;
+                    delete transaction;
```

Comment:
Replace with 
```cpp
transaction.reset()
``` 
This will call `std::shared_ptr::reset()` which will delete the memory and set the smart pointer back to `nullptr`.